examples/analog_hamiltonian_simulation/06_Solving_SAT_with_Differential_programming.ipynb

```diff
@@ -0,0 +1,872 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #3.        return p[0] * (1-(1-jnp.sin(jnp.pi * t/time_max)**2)**(p[1]/2))
</pre><p>Why parametrize the amplitude time series, if we are not optimizing its parameters? Let's remove this unnecessary complication.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
The reason lies in [qml.pulse.rydberg_drive](https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.rydberg_drive.html#qml-pulse-rydberg-drive), where amplitude can only either be a constant (float), or a function (callable). We could make it `ryd_amplitude(t)` but that is not much different from what it is now I think. It also opens the door to users if they want to optimize amplitude.